src/sidebar/test/oauth-auth-test.js

```diff
@@ -359,6 +353,33 @@ describe('sidebar.oauth-auth', function () {
       fakeSettings.services = [];
     });
 
+    [{
+      // User is logged-in on the publisher's website.
+      authority: 'publisher.org',
+      grantToken: 'a.jwt.token',
+      expectedToken: 'firstAccessToken',
+    },{
+      // User is anonymous on the publisher's website.
+      authority: 'publisher.org',
+      grantToken: null,
+      expectedToken: null,
```

Comment:
Do we need to add more cases to the `tokenGetter()` tests above this? In `describe('#tokenGetter'` I see `it('should request an access token if a grant token was provided',` but that's all. Looking at the code I think there are a number of cases to test, seems a likely candidate for a parametrized test like this one:

1. No service config at all
2. Service config contains no grantToken
3. Service config contains a grantToken but it's null. I believe other falsey values e.g. `false` may behave the same as null too.
4. Service config contains a truthy grantToken

Case 2 is also tested currently, in `it('should return null if no grant token was provided'`, but it's way further down in the file than `it('should request an access token if a grant token was provided',`, may be worth bringing them together in one parametrized test and adding additional cases.